src/Warnings.hpp

```diff
@@ -142,6 +142,7 @@ THE SOFTWARE.
     _Pragma("clang diagnostic ignored \"-Wunused-parameter\"") \
     _Pragma("clang diagnostic ignored \"-Wused-but-marked-unused\"") \
     _Pragma("clang diagnostic ignored \"-Wweak-vtables\"") \
+    _Pragma("clang diagnostic ignored \"-Wreserved-identifier\"") \
```

Comment:
`-Wreserved-identifier` is a somewhat recent warning, so older versions of clang might not know about it. Please define an `ABL_PRAGMA_CLANG_DIAGNOSTIC_IGNORED_RESERVER_IDENTIFIER` macro, similar to `ABL_PRAGMA_CLANG_DIAGNOSTIC_IGNORED_RESERVED_ID_MACRO`, so we're sure that code still builds with older versions of clang.